src/buckets/components/createBucketForm/MiniFileDnd.tsx

```diff
@@ -0,0 +1,190 @@
+ *
+ *  There can be multiple of these per page.
+ *  (The existing drag and drop components, there can
+ *  only be one per page, and there needs to be multiple of these per page
+ *  for the parent component's needs)
+ *
+ *  The style changes when a file is hovering over this component,
+ *  to show that it is active
+ *
+ * read here for drag and drop file api:
+ * https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API
+ * https://developer.mozilla.org/en-US/docs/Web/API/HTML_Drag_and_Drop_API/File_drag_and_drop
+ * */
+export const MiniFileDnd: FC<Props> = ({
+  allowedExtensions,
+  allowedTypes,
+  handleFileUpload,
+  setErrorState,
+  alreadySetFileName,
+}) => {
+  const [fileName, setFileName] = useState(alreadySetFileName)
+  const [dropAreaActive, setDropAreaActive] = useState(false)
+  const [hasError, setHasError] = useState(null)
+
+  function dragOverHandler(ev) {
+    setDropAreaActive(true)
+
+    // Prevent default behavior (Prevent file from being opened)
+    ev.preventDefault()
+  }
+
+  function dragLeaveHandler(ev) {
+    setDropAreaActive(false)
+    ev.preventDefault()
+  }
+
+  const setError = (hasError: boolean, message?: string) => {
+    setHasError(hasError)
+
+    // push error state up to the parent for better displaying
+    setErrorState(hasError, message)
+  }
+
+  const processFile = file => {
+    const reader = new FileReader()
+    reader.readAsText(file)
+    reader.onload = () => {
+      const fileName = file.name
+      handleFileUpload(reader.result as string, fileName)
+
+      setFileName(fileName)
+      setError(false)
+    }
+  }
+
+  const handleFileClick = (e: any): void => {
+    const file: File = e.currentTarget.files[0]
+
+    if (!file) {
+      return
+    }
+
+    e.preventDefault()
+    e.stopPropagation()
+
+    // don't need to see if the file type is valid; the input filters it already for us
+    processFile(file)
+  }
+
+  const dropHandler = (e: any): void => {
+    const file = e.dataTransfer.files[0]
+    const fileType = file.type
+
+    if (!file) {
+      return
+    }
+    e.preventDefault()
+    e.stopPropagation()
+
+    if (allowedTypes.includes(fileType)) {
+      processFile(file)
+    } else {
+      const niceTypes = setGrammar(allowedTypes)
+      const fileTypeInfo = fileType ? ` (${fileType})` : ''
+
+      setError(
+        true,
+        `Disallowed file type${fileTypeInfo}, you can only upload files with the type of ${niceTypes}.`
+      )
+    }
+
+    setDropAreaActive(false)
+  }
+
+  const inputEl = useRef(null)
+
+  const inputElement = (
+    <input
+      type="file"
+      data-testid="drag-and-drop--input"
+      ref={inputEl}
+      className="drag-and-drop--input"
+      accept={allowedExtensions}
+      onChange={handleFileClick}
+    />
+  )
+
+  // it gets clicked programmatically since the input itself has display='none'
+  const handleFileOpen = (): void => {
+    inputEl.current.click()
```

Comment:
do we need to check if `inputEl` has a `current` element? or is it pretty much impossible to run the `handleFileOpen` function if the component isn't mounted in the DOM?

Reply:
correct, cannot run the function if it isn't mounted